programs/clearing_house/src/math/liquidation.rs

```diff
@@ -295,3 +299,126 @@ pub fn calculate_cumulative_deposit_interest_delta_to_resolve_bankruptcy(
+
+    let cost_basis_above_mean = if deleverage_user_stats.base_asset_amount != 0 {
+        let quote_entry = (-market.amm.quote_entry_amount_long)
+            .safe_add(market.amm.quote_entry_amount_short)?
+            .safe_add(
+                loss_to_socialize
+                    .safe_mul(deleverage_user_stats.base_asset_amount.signum().cast()?)?,
+            )?;
+
+        let mean_entry_basis = if base_amount != 0 {
+            (quote_entry)
+                .safe_mul(PRICE_PRECISION_I128)?
+                .safe_mul(AMM_TO_QUOTE_PRECISION_RATIO_I128)?
+                .safe_div(base_amount)?
+        } else {
+            0
+        };
+
+        let user_entry_basis: i128 = if deleverage_user_stats.base_asset_amount != 0 {
+            PRICE_PRECISION_I128
+                .safe_mul(-(deleverage_user_stats.quote_entry_amount).cast()?)?
+                .safe_mul(AMM_TO_QUOTE_PRECISION_RATIO_I128)?
+                .safe_div(deleverage_user_stats.base_asset_amount.cast()?)?
+        } else {
+            0_i128
+        };
+
+        let basis_above_mean: i128 = if deleverage_user_stats.base_asset_amount > 0
+            && oracle_price > user_entry_basis.cast()?
+        {
+            mean_entry_basis.safe_sub(user_entry_basis.cast()?)?
+        } else if deleverage_user_stats.base_asset_amount < 0
+            && oracle_price < user_entry_basis.cast()?
+        {
+            -(mean_entry_basis.safe_sub(user_entry_basis.cast()?)?)
+        } else {
+            -1
+        };
+
+        dlog!(
+            quote_entry,
+            base_amount,
+            mean_entry_basis,
+            user_entry_basis,
+            basis_above_mean
+        );
+
+        basis_above_mean
+    } else {
+        0
+    };
+
+    let profit_above_mean = if deleverage_user_stats.base_asset_amount == 0 {
+        if deleverage_user_stats.quote_asset_amount != 0 {
+            let mean_quote = (market.amm.base_asset_amount_short)
+                .safe_add(market.amm.base_asset_amount_long)?
+                .safe_mul(oracle_price.cast()?)?
+                .safe_div(PRICE_TIMES_AMM_TO_QUOTE_PRECISION_RATIO_I128)?
+                .safe_add(
+                    market
+                        .amm
+                        .quote_asset_amount_long
+                        .safe_add(market.amm.quote_asset_amount_short)?
+                        .safe_sub(market.amm.cumulative_social_loss)?
+                        .safe_add(loss_to_socialize)?,
+                )?
+                .safe_div(market.number_of_users.max(1).cast()?)?;
+
+            dlog!(mean_quote);
+
+            -(mean_quote.safe_sub(deleverage_user_stats.quote_asset_amount.cast()?)?)
+        } else {
+            0
+        }
+    } else if cost_basis_above_mean == 0 {
+        loss_to_socialize
+            .abs()
+            .safe_div(market.number_of_users.max(1).cast()?)?
+            .max(1)
```

Comment:
why this extra max?

Reply:
if `loss_to_socialize` is less than `market.number_of_users`. and if `number_of_users` is 0 dont want to divide by 0